datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -385,6 +386,95 @@ def save_node(
     session.refresh(node.current)
 
 
+def copy_to_new_node(
+    session: Session,
+    node: Node,
+    new_name: str,
+    current_user: Optional[User] = None,
+) -> Node:
+    """
+    Copies the existing node to a new node with a new name.
+    """
+    old_revision = node.current
+    new_node = Node(
+        name=new_name,
+        type=node.type,
+        display_name=node.display_name,
+        namespace=".".join(new_name.split(".")[:-1]),
+        current_version=node.current_version,
+        created_at=node.created_at,
+        deactivated_at=node.deactivated_at,
+        tags=node.tags,
+        missing_table=node.missing_table,
+    )
+    new_revision = NodeRevision(
+        name=new_name,
+        display_name=old_revision.display_name,
+        type=old_revision.type,
+        description=old_revision.description,
+        query=old_revision.query,
+        mode=old_revision.mode,
+        version=old_revision.version,
+        node=new_node,
+        catalog=old_revision.catalog,
+        schema_=old_revision.schema_,
+        table=old_revision.table,
+        required_dimensions=[col.copy() for col in old_revision.required_dimensions],
+        metric_metadata=old_revision.metric_metadata,
+        cube_elements=list(old_revision.cube_elements),
+        status=old_revision.status,
+        parents=old_revision.parents,
+        missing_parents=[
+            MissingParent(name=missing_parent.name)
+            for missing_parent in old_revision.missing_parents
+        ],
+        columns=[col.copy() for col in old_revision.columns],
+        # TODO: availability and materializations are missing here  # pylint: disable=fixme
+        lineage=old_revision.lineage,
+    )
+    new_revision.dimension_links = [
```

Comment:
What's the reason `dimension_links` are set outside the previous call? This looks similar to the `missing_parents` but maybe I am missing something?

Reply:
Good point, I thought it wouldn't work because we had to set `node_revision` to point the new revision, but I think `back_populates` should actually take care of that. Let me give it another try.